### PR TITLE
Flaskapi merge

### PR DESCRIPTION
**Pull request by SavSanta**
# Description

This pull requests implements a functional API backend using the Flask's builtin test webserver, a test SQLite3 database†, and a DOCKERFILE that can be used to build and run an image using alpine linux.

There are few routes for calls established and can be interfaced with  which is briefly documented at the main index page once the server is stood up as well as on this github branch documentation using the `curl` program (available typically on Linux/Unix/Mac. Windows users should be able to similarly utilize something like Invoke-WebRequest).

† Note that the SQLite3 database backend files are not attached as it is meant for usage as testing and we're still missing one contributor's code for NTEE normalized names. It is attached for usage here if you wish to download and add it to the project directory.

### CODE REVIEW

**Comment by SavSanta** [comment]
SQLite3 TEST Database -- 
[database-devel.sqlite3.zip](https://github.com/CodeForBaltimore/CitySync/files/4295282/database-devel.sqlite3.zip)

File: README.md [hunk 1]
```diff
@@ -1,42 +1,35 @@
...
 To run this locally the following software is required:
-*  [Python](https://www.python.org/)
-*  [Docker](https://docker.com) *optional*
-
-If on a Mac you can install Node and npm via Homebrew with `brew install python`
+*  [Python](https://www.python.org/) (version minimum 3.6)
+*  [Flask](https://palletsprojects.com/p/flask/)
+*  [Flask-Marshmallow](https://flask-marshmallow.readthedocs.io)
+*  [Flask-SQLAlchemy](https://flask-sqlalchemy.palletsprojects.com)
+*  [Docker](https://docker.com) (*for optional container image usage*)
```
Comment (stoopidJSON) [inline README.md:11]: Maybe add a small bit about adding the SQL db

File: citysync/requirements.txt [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+Flask-Migrate==2.5.2
+Flask-SQLAlchemy==2.4.1
+Flask-WTF==0.14.2
+itsdangerous==1.1.0
+Jinja2==2.11.0
+Mako==1.1.1
+MarkupSafe==1.1.1
+marshmallow==3.3.0
+marshmallow-sqlalchemy==0.21.0
+pkg-resources==0.0.0
```
Comment (stoopidJSON) [inline citysync/requirements.txt]: This is causing an error during Docker build `ERROR: Could not find a version that satisfies the requirement pkg-resources==0.0.0`

**Review by stoopidJSON — CHANGES_REQUESTED** [review]
I'm not able to run this yet due to the error noted in the comments, but this looks great so far!

File: citysync/requirements.txt (hunk 2 above)
Comment (SavSanta) [reply citysync/requirements.txt]: Thatcan be removed it was generated via `pip freeze` which I guess added all self-contained dependencies. I just ran it on an EC2 instance and it works with that line removed

File: README.md (hunk 1 above)
Comment (SavSanta) [reply README.md:11]: The "test" database is not intended to be shipped. This is for dev testing only. Someone will need to figure out the proper seeding mechanism required once someone moves to an official database. Hopefully, no one is intending to run off a SQLITE3 on an actual production system due to concurrent writes causing DBLocks

Comment (stoopidJSON) [reply README.md:11]: Fair enough, but it is a requirement to run the application at this time. I would just suggest adding something to explain that for now, but that's your call.

**Review by stoopidJSON — APPROVED** [review]
A minor README addition could be added, but otherwise, it runs locally. Looks good to me!

**Pull request merged**